src/Components/AddProject/AddProject.tsx

```diff
@@ -0,0 +1,114 @@
+// nadia.mohamed.taha166@gmail.com
+// @Password123!
+import axios from "axios";
+import { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { AuthContext } from "./../../Context/AuthContext";
+import { ToastContext } from "../../Context/ToastContext";
+import { SubmitHandler,useForm } from "react-hook-form";
+
+const AddProject: React.FC = () => {
+  const { baseUrl, requestHeaders }: any = useContext(AuthContext);
+  const { getToastValue }: any = useContext(ToastContext);
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+  interface FormValues {
+    title: string;
+    description: string;
+  }
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>();
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    setIsLoading(true)
+    axios
+      .post(`${baseUrl}/Project`, data, { headers: requestHeaders })
+      .then((response) => {
+
+        setIsLoading(false)
```

Comment:
Consider using finally(() => setIsLoading(false)) after the promise to consistently reset the loading state to false, avoiding redundancy in both the then and catch blocks.